File: test_connect.py
import psycopg

conn = psycopg.connect(
    "host=127.0.0.1 port=5432 dbname=test user=test password=test",
    connect_timeout=3,
)
conn.autocommit = True
cur = conn.cursor()

print("== DROP TABLE (if present) ==")
try:
    cur.execute("DROP TABLE users")
    print("dropped existing")
except psycopg.errors.SyntaxError:
    print("none to drop")

print("== CREATE TABLE ==")
cur.execute("CREATE TABLE users (name TEXT, age INT)")
print("ok")

print("== INSERT ==")
cur.execute("INSERT INTO users (name, age) VALUES ('Alice', 30)")
cur.execute("INSERT INTO users (name, age) VALUES ('Bob', 25)")
print("ok")

print("== SELECT ==")
cur.execute("SELECT * FROM users")
for row in cur.fetchall():
    print(" ", row)

print("== SELECT WHERE ==")
cur.execute("SELECT name FROM users WHERE age > 26")
for row in cur.fetchall():
    print(" ", row)

print("== UPDATE ==")
cur.execute("UPDATE users SET age = 31 WHERE name = 'Alice'")
print("rows updated:", cur.rowcount)

print("== SELECT AGAIN ==")
cur.execute("SELECT id, name, age FROM users")
for row in cur.fetchall():
    print(" ", row)

print("== DELETE ==")
cur.execute("DELETE FROM users WHERE name = 'Bob'")
print("rows deleted:", cur.rowcount)

print("== FINAL ==")
cur.execute("SELECT * FROM users")
for row in cur.fetchall():
    print(" ", row)

conn.close()
print("DONE")

File: test_connect.php
<?php

$dsn = 'pgsql:host=127.0.0.1;port=5432;dbname=test';
$user = 'test';
$pass = 'test';

try {
    $pdo = new PDO($dsn, $user, $pass, [
        PDO::ATTR_TIMEOUT => 3,
    ]);
} catch (PDOException $e) {
    echo "could not connect: ", $e->getMessage(), "\n";
    exit(1);
}

$pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

echo "== DROP TABLE (if present) ==\n";
try {
    $pdo->exec("DROP TABLE users");
    echo "dropped existing\n";
} catch (PDOException $e) {
    echo "none to drop\n";
}

echo "== CREATE TABLE ==\n";
$pdo->exec("CREATE TABLE users (name TEXT, age INT)");
echo "ok\n";

echo "== CREATE TABLE IF NOT PRESENT ==\n";
$pdo->exec("CREATE TABLE IF NOT PRESENT users (name TEXT, age INT)");
echo "ok\n";

echo "== INSERT ==\n";
$pdo->exec("INSERT INTO users (name, age) VALUES ('Alice', 30)");
$pdo->exec("INSERT INTO users (name, age) VALUES ('Bob', 25)");
echo "ok\n";

echo "== SELECT ==\n";
$stmt = $pdo->query("SELECT * FROM users");
foreach ($stmt as $row) {
    echo "  ";
    print_r($row);
}

echo "== SELECT WHERE ==\n";
$stmt = $pdo->query("SELECT name FROM users WHERE age > 26");
foreach ($stmt as $row) {
    echo "  ";
    print_r($row);
}

echo "== UPDATE ==\n";
$n = $pdo->exec("UPDATE users SET age = 31 WHERE name = 'Alice'");
echo "rows updated: ", $n, "\n";

echo "== SELECT AGAIN ==\n";
$stmt = $pdo->query("SELECT id, name, age FROM users");
foreach ($stmt as $row) {
    echo "  ";
    print_r($row);
}

echo "== DELETE ==\n";
$n = $pdo->exec("DELETE FROM users WHERE name = 'Bob'");
echo "rows deleted: ", $n, "\n";

echo "== DONE ==\n";
